core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -165,19 +165,19 @@ object ZkUtils {
   def getDeleteTopicPath(topic: String): String =
     DeleteTopicsPath + "/" + topic
 
-  // Parses without deduplicating keys so the data can be checked before allowing reassignment to proceed
   def parsePartitionReassignmentData(jsonData: String): Map[TopicAndPartition, Seq[Int]] = {
+    val parseResult = Json.parseTo(jsonData, classOf[PartitionAssignment])
+
+    if (parseResult.isLeft)
+      throw new ConfigException(s"Invalid reassignment config: ${parseResult.left}")
+
+    val assignments = parseResult.right
```

Comment:
You should use pattern matching to retrieve `left` or `right` so that you can avoid the unsafe `get` method below.